patches/log.py

```diff
@@ -0,0 +1,132 @@
+            if stack_info:
+                sio = io.StringIO()
+                sio.write('Stack (most recent call last):\n')
+                traceback.print_stack(f, file=sio)
+                sinfo = sio.getvalue()
+                if sinfo[-1] == '\n':
+                    sinfo = sinfo[:-1]
+                sio.close()
+            rv = (co.co_filename, f.f_lineno, co.co_name, sinfo)
+            break
+        return rv
+
+
+_srcfile = os.path.normcase(_Logger.findCaller.__code__.co_filename)
+_ignore_srcfiles = (_srcfile, logging._srcfile)
+
+class Log:
+    """ Helper aggregate for all things related to logging activity. """
+
+    _GLOBAL_LOG = logging.getLogger("")
+    _initialized = False
+
+    # These are convenience helpers. For performance, consider saving Log.get_logger() and using that
+    @staticmethod
+    def critical(msg, *args, **kwargs):
+        Log._GLOBAL_LOG.critical(msg, *args, **kwargs)
+
+    @staticmethod
+    def fatal(msg, *args, **kwargs):
+        Log._GLOBAL_LOG.fatal(msg, *args, **kwargs)
+
+    @staticmethod
+    def exception(msg, *args, **kwargs):
+        Log._GLOBAL_LOG.exception(msg, *args, **kwargs)
+
+    @staticmethod
+    def error(msg, *args, **kwargs):
+        Log._GLOBAL_LOG.error(msg, *args, **kwargs)
+
+    @staticmethod
+    def warn(msg, *args, **kwargs):
+        Log._GLOBAL_LOG.warn(msg, *args, **kwargs)
+
+    @staticmethod
+    def warning(msg, *args, **kwargs):
+        Log._GLOBAL_LOG.warning(msg, *args, **kwargs)
+
+    @staticmethod
+    def debug(msg, *args, **kwargs):
+        Log._GLOBAL_LOG.debug(msg, *args, **kwargs)
+
+    @staticmethod
+    def info(msg, *args, **kwargs):
+        Log._GLOBAL_LOG.info(msg, *args, **kwargs)
+
+    @staticmethod
+    def set_level(loglevel):
+        if isinstance(loglevel, int):
+            Log._GLOBAL_LOG.setLevel(loglevel)
+            return
+        elif isinstance(loglevel, str):
+            # idea from https://docs.python.org/3.5/howto/logging.html#logging-to-a-file
+            numeric_level = getattr(logging, loglevel.upper(), None)
+            if isinstance(numeric_level, int):
+                Log._GLOBAL_LOG.setLevel(numeric_level)
+                return
+
+        raise ValueError('Invalid log level: %s' % loglevel)
+
+    @staticmethod
+    def _static_init():
+        if Log._initialized:
+            return
+
+        logging.setLoggerClass(_Logger)
+        # The root logger's type is unfortunately (and surprisingly) not affected by
+        # `setLoggerClass`. Monkey patch it instead. TODO(vimota): Remove this, see the TODO
+        # associated with _Logger.
+        logging.RootLogger.findCaller = _Logger.findCaller
+        log_to_file = _LOG_TO_FILE_ENV.lower() in ("yes", "true", "t", "1") if _LOG_TO_FILE_ENV is not None else True
+        if log_to_file:
+            handler = logging.FileHandler(filename='/tmp/kaggle.log', mode='w')
+        else:
+            handler = logging.StreamHandler()
```

Comment:
If `KAGGLE_LOG_TO_FILE` is set to false and we stream the logs to stdin/err, does that mean that user will see it in the editor console? 

Do you intend to set KAGGLE_LOG_TO_FILE to false in dev for easier debugging?

Reply:
I believe so, and yes, the idea would be to be able to develop locally more easily by printing things out while they're running! :)